### PR TITLE
[AAE-11480] - Fix required error is displayed for a non required drop…

### PR DESCRIPTION
**Pull request by arditdomi**
…down

**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-11480

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by DenysVuika — APPROVED** [review]

File: lib/core/src/lib/form/components/widgets/core/form-field.model.spec.ts [hunk 1]
```diff
@@ -531,18 +531,21 @@ describe('FormFieldModel', () => {
...
+    it('should set the value as null for a dropdown field that has the None value selected', () => {
         const form = new FormModel();
         const field = new FormFieldModel(form, {
             id: 'dropdown-1',
             type: FormFieldTypes.DROPDOWN
         });
 
         field.value = 'empty';
-        expect(form.values['dropdown-1']).toBe(undefined);
+        expect(form.values['dropdown-1']).toBe(null);
```
Comment (mauriziovitale) [inline lib/core/src/lib/form/components/widgets/core/form-field.model.spec.ts:542]: why null and not undefined ?

File: lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.html [hunk 2]
```diff
@@ -7,6 +7,9 @@
     </div>
     <div>
         <mat-form-field>
+            <mat-label *ngIf="hasDefaultOption(list$ | async)">
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.html:10]: I think u need an alias or u will subscribe 2 times

Comment (mauriziovitale) [reply lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.html:10]: https://github.com/Alfresco/alfresco-apps/pull/4592#discussion_r1034696923

File: lib/core/src/lib/form/components/widgets/core/form-field.model.ts [hunk 3]
```diff
@@ -372,7 +378,7 @@ export class FormFieldModel extends FormWidgetModel {
 
                 if (typeof this.value === 'string') {
                     if (this.value === 'empty' || this.value === '') {
-                        delete this.form.values[this.id];
+                        this.form.values[this.id] = null;
```
Comment (mauriziovitale) [inline lib/core/src/lib/form/components/widgets/core/form-field.model.ts:381]: can we use undefined as in the other places ? 
https://github.com/Alfresco/alfresco-ng2-components/pull/8021/files#diff-ae592ea8f0e66bf49f5dcd46b7d6ad926e83b70fb3aed81c146cad30a15fc9a7R150

**Review by mauriziovitale — CHANGES_REQUESTED** [review]
comments

File: lib/core/src/lib/form/components/widgets/core/form-field.model.ts (hunk 3 above)
Comment (arditdomi) [reply lib/core/src/lib/form/components/widgets/core/form-field.model.ts:381]: The API expects null in order to unset the value of a dropdown field. undefined wouldn't work in this case

File: lib/core/src/lib/form/components/widgets/core/form-field.model.spec.ts (hunk 1 above)
Comment (arditdomi) [reply lib/core/src/lib/form/components/widgets/core/form-field.model.spec.ts:542]: Same as https://github.com/Alfresco/alfresco-ng2-components/pull/8021#discussion_r1037042264

**Review by mauriziovitale — APPROVED** [review]

**Pull request merged**